Formula/pytorch.rb

```diff
@@ -48,14 +40,22 @@ def install
 
     resource("fbgemm").stage(buildpath/"third_party/fbgemm")
 
+    # Remove with next release
+    inreplace "cmake/Dependencies.cmake",
+      'if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 13.0.0)',
+      'if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")'
+
```

Comment:
I think instead we need to backport the part of https://github.com/pytorch/pytorch/commit/d7e6aaa59b4660876d3d9bbee50113fad2c0b115 that changes 
```
target_compile_options(asmjit -Wno-unused-but-set-variable)
```
to
```
target_compile_options_if_supported(asmjit -Wno-unused-but-set-variable)
```
That will make sure that flag is only added if the compile supports the flag, where as in the current release it uses the flag unconditionally.

Reply:
I thought about doing that, but unfortunately the `target_compile_options_if_supported` macro is not present in the 1.12.1 release! It's a custom macro that was committed after the release was created. So we would have to patch that function back in as well, which seems like a bigger change than necessary.

I think maybe we can just delete these lines entirely? That's what I'm trying now in my latest commit.

Reply:
Nope, patches aren't allowed in new formulae, and brew-test-bot sees this as a new formula (even though it's a rename, which is different). So I'm going to try patching it to `if(FALSE)` instead for the same effect.

Reply:
It looks like this worked now and just timed out, though we should only do the `inreplace` if the macOS version is less than or equal to Catalina.

Reply:
Why less than or equal to Catalina? Also, could you please add the "long build" label to this pull request, so that it doesn't time out for the next CI build?

Reply:
Actually let's do a CI run to see if all platforms can build with the `if(FALSE)` fix used.  We'll be able to drop it in the next release anyway.

Reply:
CI worked at last!  Great job hunting down that fix, and thanks for your patience.  This PR was beset by multiple unrelated problems that were non-trivial to fix but I think we're ready to go.  I will merge this tomorrow if there has been no other maintainer feedback as the other changes here are pretty straight forward.